integrations/aws/main.py

```diff
@@ -69,19 +73,37 @@ async def _handle_global_resource_resync(
 async def resync_all(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
     if kind in iter(ResourceKindsWithSpecialHandling):
         return
+
     await update_available_access_credentials()
     is_global = is_global_resource(kind)
-    async for credentials in get_accounts():
-        if is_global:
-            async for batch in _handle_global_resource_resync(kind, credentials):
-                yield batch
-        else:
-            async for session in credentials.create_session_for_each_region():
-                try:
-                    async for batch in resync_cloudcontrol(kind, session):
-                        yield batch
-                except Exception:
-                    continue
+    semaphore = asyncio.BoundedSemaphore(MAX_CONCURRENT_TASKS)
+
+    async def handle_account(
```

Comment:
Extract this function.

Reply:
The main purpose of the resync_all is to control the rate and the parallelism of the handle_account. Also, rename this handle account to something more verbose.

Reply:
can you come again, this isn't clear to me ?